benchmarks/main.ts

```diff
@@ -0,0 +1,325 @@
+import * as windmill from "https://deno.land/x/windmill@v1.37.0/mod.ts";
+import * as api from "https://deno.land/x/windmill@v1.37.0/windmill-api/index.ts";
+import { InfluxDB, Point, HttpError } from "npm:@influxdata/influxdb-client";
+import { BucketsAPI, OrgsAPI } from "npm:@influxdata/influxdb-client-apis";
+
+async function login(
+  config: api.Configuration,
+  email: string,
+  password: string
+): Promise<string> {
+  return await new windmill.UserApi(config).login({
+    email: email,
+    password: password,
+  });
+}
+
+await new Command()
+  .name("windmill-bench")
+  .description("Run Benchmark to measure throughput of windmill.")
+  .version("v0.0.0")
+  .option("--host <url:string>", "The windmill host to benchmark.", {
+    default: "http://127.0.0.1/",
+  })
+  .option("-j --jobs <jobs:number>", "The number of jobs to run at once.", {
+    default: 1,
+  })
+  .option(
+    "-s --seconds <seconds:number>",
+    "How long to run the benchmark for (in seconds).",
+    {
+      default: 30,
+    }
+  )
+  .option("-e --email <email:string>", "The email to use to login.")
+  .option("-p --password <password:string>", "The password to use to login.")
+  .env(
+    "WM_TOKEN=<token:string>",
+    "The token to use when talking to the API server. Preferred over manual login."
+  )
+  .option(
+    "-t --token <token:string>",
+    "The token to use when talking to the API server. Preferred over manual login."
+  )
+  .env(
+    "WM_WORKSPACE=<workspace:string>",
+    "The workspace to spawn scripts from."
+  )
+  .option(
+    "-w --workspace <workspace:string>",
+    "The workspace to spawn scripts from.",
+    { default: "starter" }
+  )
+  .option("-m --metrics <metrics:string>", "The url to scrape metrics from.", {
+    default: "http://localhost:8001/metrics",
+  })
+  .option("--influx-host <influx_host:string>", "The influx url to write to.")
+  .option("--influx-token <influx_token:string>", "The influx token to use.")
+  .option("--influx-org <influx_org:string>", "The influx org to write to.")
+  .option(
+    "--influx-bucket <influx_bucket:string>",
+    "The influx bucket to write to. Everything in the bucket will be deleted!!"
+  )
+  .option(
+    "--export-json <export_json:string>",
+    "If set, exports will be into a JSON file."
+  )
+  .option(
+    "--exports [exports...:string]",
+    "Mark metrics (without label) to export."
+  )
+  .arguments("[domain]")
+  .action(
+    async ({
+      host,
+      jobs,
+      seconds,
+      email,
+      password,
+      token,
+      workspace,
+      metrics,
+      influxHost,
+      influxToken,
+      influxOrg,
+      influxBucket,
+      exportJson,
+      exports,
+    }) => {
+      const metrics_worker = new Worker(
+        new URL("./scraper.ts", import.meta.url).href,
+        {
+          type: "module",
+        }
+      );
+
+      const export_map: Map<
+        string,
+        Map<string, { val: number; samples: number }>
+      > = new Map();
```

Comment:
I usually use Record and very seldom Map. Not that there is anything wrong with Map but curious to know if this is an habit from another language or you had strong reasons for it.

Reply:
Just applied Record in a few places where I used the more complicated indexer syntax, had forgotten about that type.
I guess Map can be replaced by Record[] sometimes, but I usually just prefer Map where keys are unique and I want to iterate over it.
Here it's awkward a few times since I have to use `new Map(Object.entries(obj))` and `Object.fromEntries` to go from/to map a few times.